viabel/objectives.py

```diff
@@ -304,6 +304,9 @@ def f_model(x):
                 elbo_gsamps_tilde_centered = np.column_stack([hvps, dLds - dLds_mu])
                 g_hat_rv = np.mean(g_hat_rprm_grad - elbo_gsamps_tilde_centered, axis=0)
 
+            # TODO : IDE reported g_hat_rv can be called before assigned since g_hat_rv is only assigned under the if
```

Comment:
Add an else statement that raises a RuntimeException (or similar) with an informative error message.